src/Controller/DemoController.php

```diff
@@ -36,55 +36,68 @@ final class DemoController extends AbstractController
      */
     private $rectorRunRepository;
 
-    public function __construct(RectorProcessRunner $rectorProcessRunner, RectorRunRepository $rectorRunRepository)
-    {
+    /**
+     * @var FileDiffCleaner
+     */
+    private $fileDiffCleaner;
+
+    public function __construct(
+        RectorProcessRunner $rectorProcessRunner,
+        RectorRunRepository $rectorRunRepository,
+        FileDiffCleaner $fileDiffCleaner
+    ) {
         $this->rectorProcessRunner = $rectorProcessRunner;
         $this->rectorRunRepository = $rectorRunRepository;
+        $this->fileDiffCleaner = $fileDiffCleaner;
     }
 
     /**
-     * @Route(path="demo/{id}", name="demo_detail", methods={"GET", "POST"})
+     * @Route(path="demo/{rectorRun}", name="demo_detail", methods={"GET", "POST"})
      * @Route(path="demo", name="demo", methods={"GET", "POST"})
      */
-    public function __invoke(Request $request): Response
+    public function __invoke(Request $request, ?RectorRun $rectorRun = null): Response
     {
-        $formData = new RectorRunFormData();
-        $rectorRun = null;
-        $rectorRunId = $request->attributes->get('id');
-
-        $demoFileContent = FileSystem::read(__DIR__ . '/../../data/DemoFile.php');
-        $demoConfig = FileSystem::read(__DIR__ . '/../../data/demo-config.yaml');
-
-        $formData->setContent($demoFileContent);
-        $formData->setConfig($demoConfig);
-
-        if ($rectorRunId) {
-            $rectorRun = $this->rectorRunRepository->get(Uuid::fromString($rectorRunId));
-
-            $formData->setContent($rectorRun->getContent());
-            $formData->setConfig($rectorRun->getConfig());
-        }
+        $formData = $this->createDemoFormData($rectorRun);
 
-        $form = $this->createForm(RectorRunFormType::class, $formData);
+        $form = $this->createForm(DemoFormType::class, $formData);
         $form->handleRequest($request);
 
         if ($form->isSubmitted() && $form->isValid()) {
             return $this->processFormAndReturnRoute($form);
         }
 
         return $this->render('homepage/demo.twig', [
-            'form' => $form->createView(),
-            'rector_run' => $rectorRun,
+            'demo_form' => $form->createView(),
+            'rector_run' => $rectorRun ?? null,
         ]);
     }
 
+    private function createDemoFormData(?RectorRun $rectorRun): DemoFormData
+    {
+        $formData = new DemoFormData();
+
+        if ($rectorRun) {
+            $formData->setContent($rectorRun->getContent());
```

Comment:
I would love if it would be possible to have class `DemoFormData` without setters and only pass parameters via constructor. Can we do it?